config/config.go

```diff
@@ -15,138 +15,474 @@ package config
+
+	// Notification configurations to use for this AggregationRule, referenced
+	// by their name.
+	NotificationConfigs []string `yaml:"notification_configs"`
+
+	// Notify when resolved.
+	SendResolved bool `yaml:"send_resolved"`
+
+	// Catches all undefined fields and must be empty after parsing.
+	XXX map[string]interface{} `yaml:",inline"`
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (r *AggrRule) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*r = DefaultAggrRule
+	type plain AggrRule
+	if err := unmarshal((*plain)(r)); err != nil {
+		return err
 	}
+	if len(r.NotificationConfigs) == 0 {
+		return fmt.Errorf("aggregation rule needs at least one notification config")
+	}
+	return checkOverflow(r.XXX, "aggregation rule")
+}
 
-	return nil
+// A regex-based label filter used in aggregations.
+type Filter struct {
+	Name  string  `yaml:"name"`
+	Regex *Regexp `yaml:"regex"`
+
+	// Catches all undefined fields and must be empty after parsing.
+	XXX map[string]interface{} `yaml:",inline"`
 }
 
-func filtersFromPb(filters []*pb.Filter) manager.Filters {
-	fs := make(manager.Filters, 0, len(filters))
-	for _, f := range filters {
-		fs = append(fs, manager.NewFilter(f.GetNameRe(), f.GetValueRe()))
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (f *Filter) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	type plain Filter
+	if err := unmarshal((*plain)(f)); err != nil {
+		return err
 	}
-	return fs
+	return checkOverflow(f.XXX, "aggregation rule")
 }
 
-// AggregationRules returns all the AggregationRules in a Config object.
-func (c Config) AggregationRules() manager.AggregationRules {
-	rules := make(manager.AggregationRules, 0, len(c.AggregationRule))
-	for _, r := range c.AggregationRule {
-		rate := time.Duration(r.GetRepeatRateSeconds()) * time.Second
-		if rate < minimumRepeatRate {
-			rate = minimumRepeatRate
-		}
-		rules = append(rules, &manager.AggregationRule{
-			Filters:                filtersFromPb(r.Filter),
-			RepeatRate:             rate,
-			NotificationConfigName: r.GetNotificationConfigName(),
-		})
-	}
-	return rules
-}
-
-// InhibitRules returns all the InhibitRules in a Config object.
-func (c Config) InhibitRules() manager.InhibitRules {
-	rules := make(manager.InhibitRules, 0, len(c.InhibitRule))
-	for _, r := range c.InhibitRule {
-		sFilters := filtersFromPb(r.SourceFilter)
-		tFilters := filtersFromPb(r.TargetFilter)
-		rules = append(rules, &manager.InhibitRule{
-			SourceFilters: sFilters,
-			TargetFilters: tFilters,
-			MatchOn:       r.MatchOn,
-		})
-	}
-	return rules
+// Notification configuration definition.
+type NotificationConfig struct {
+	// Name of this NotificationConfig. Referenced from AggregationRule.
+	Name string `yaml:"name"`
+
+	// Notify when resolved.
+	SendResolved bool `yaml:"send_resolved"`
+
+	// Zero or more PagerDuty notification configurations.
+	PagerdutyConfigs []*PagerdutyConfig `yaml:"pagerduty_configs"`
```

Comment:
A pity that repeated elements come at a higher cost in YAML, but probably worth keeping all these repeated.